### PR TITLE
add makefile

### PR DESCRIPTION
**Pull request by tupyy**
This PR adds a simple Makefile
Signed-off-by: Cosmin Tupangiu <cosmin@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	awk 'BEGIN {FS = ":"}; {if ($$1 ~ /\./) printf("    $(COLOR_BLUE)%-21s$(RESET_COLOR) %s\n", $$1, $$2); else printf("$(COLOR_YELLOW)%-25s$(RESET_COLOR) %s\n", $$1, $$2)}'
+
+
+#####################
+# Build targets     #
+#####################
+NAME=calgo
+VERSION=0.1
+GIT_COMMIT=$(shell git rev-list -1 HEAD --abbrev-commit)
+GO_VERSION=1.17
```
Comment (rgolangh) [inline Makefile]: why not 1.18?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+COLOR_LEVEL_INFO=$(escape)[92m
+COLOR_LEVEL_WARN=$(escape)[38;5;208m
+COLOR_LEVEL_ERROR=$(escape)[91m
+COLOR_LEVEL_FATAL=$(escape)[91m
+
+define COLORIZE
+sed -u -e "s/\\\\\"/'/g; \
+s/Method\([^ ]*\)/Method$(COLOR_BLUE)\1$(RESET_COLOR)/g;        \
+s/ERROR\"\([^\"]*\)\"/error=\"$(COLOR_RED)\1$(RESET_COLOR)\"/g;  \
+s/ProductID:\s\([^\"]*\)/$(COLOR_YELLOW)ProductID: \1$(RESET_COLOR)/g;   \
```
Comment (rgolangh) [inline Makefile]: is this ProductID in use?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#help help.all: display all targets' help messages
+help.all:
+	@grep -hE '^#help|^[a-z_-]+:' $(MAKEFILE_LIST) | sed "s/#help //g" | LANG=C sort -d | \
+	awk 'BEGIN {FS = ":"}; {if ($$1 ~ /\./) printf("    $(COLOR_BLUE)%-21s$(RESET_COLOR) %s\n", $$1, $$2); else printf("$(COLOR_YELLOW)%-25s$(RESET_COLOR) %s\n", $$1, $$2)}'
+
+
+#####################
+# Build targets     #
+#####################
+NAME=calgo
```
Comment (rgolangh) [inline Makefile]: Can you move those important vars the top of this file?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+build.local: build.prepare
+	go build -mod=vendor $(BUILD_ARGS) -ldflags "-X main.CommitID=$(GIT_COMMIT) -s -w" \
+	-o $(CURDIR)/target/$(NAME) $(CURDIR)/main.go
+
+#####################
+# Check targets     #
+#####################
+
+LINT_COMMAND=golangci-lint run
+FILES_LIST=$(shell ls -d */ | grep -v -E "vendor|tools|target|client|restapi|models")
```
Comment (rgolangh) [inline Makefile]: why is this needed? most of the folders here do not exist

File: Makefile (hunk 2 above)
Comment (tupyy) [reply Makefile]: I'll remove entirely the colorize section cause it's made for logrus

File: Makefile (hunk 4 above)
Comment (tupyy) [reply Makefile]: the goal here is to run `goimports` on all the folders _except_ those in the list. I'll remove client, restapi, models and tools

**Comment by rgolangh** [comment]
I'll close this for now to keep the project small and keep the effort in the missing features
